### PR TITLE
chore(linux): Refactor scripts and packaging 🏗️

### PR DESCRIPTION
**Pull request by ermshiperete**
This change modifies the scripts and the Debian packaging to use the meson build / build.sh script for building instead of autotools.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by mcdurdin** [comment]
LGTM but why are the builds failing? :grin:

**Comment by ermshiperete** [comment]
> LGTM but why are the builds failing? grin

There's still a dependency problem. core needs to be built before ibus-keyman can be configured.

File: linux/debian/rules
Comment (mcdurdin) [inline linux/debian/rules:62]: Is it still appropriate to be setting `DESTDIR` here?

File: linux/scripts/reconf.sh
Comment (mcdurdin) [inline linux/scripts/reconf.sh:42]: Does this mean that we could build keyboardprocessor twice? Looking above it seems that `keyboardprocessor` and `keyman` never end up specified together, so perhaps this is okay, but it seems kinda funny.

**Review by mcdurdin — APPROVED** [review]
LGTM

File: linux/debian/rules
Comment (ermshiperete) [reply linux/debian/rules:62]: yes, that's needed so that the files we install end up in a dedicated directory so that the packaging system knows what to put in the package.

**Comment by ermshiperete** [comment]
Don't merge until #8112 is approved!

**Pull request merged**
